test-utils/storage.ts

```diff
@@ -0,0 +1,16 @@
+import { Storage } from "../src";
+
+const store: { [k: string]: string } = {};
+export const testStorage: Storage & { clearStore: () => void } = {
+  setItem: (key: string, data: string) => {
+    store[key] = data;
+    return Promise.resolve();
+  },
+  getItem: (key: string) => {
+    const value = store[key];
+    return Promise.resolve(value);
+  },
+  clearStore: () => {
+    for (const key in store) delete store[key];
```

Comment:
This looks good and works for tests, the only thing you could perhaps leverage is a `Map` which is typical for a quick in-memory store.

Reply:
done